### PR TITLE
Add Work from Connect() Demo

### PR DESCRIPTION
**Pull request by mhamilton723**
- Add ability to write images back to storage
- Add ability to read images from a dataframe of paths

### CODE REVIEW

File: src/plot/src/main/python/plot.py [hunk 1]
```diff
@@ -13,12 +13,12 @@
 from pyspark.sql import DataFrame
 from pyspark.sql.types import *
 
-import matplotlib.pyplot as plt
 import numpy as np
 from sklearn.metrics import confusion_matrix, roc_curve
 import itertools
 
 def confusionMatrix(df, y_col, y_hat_col, labels):
+    import matplotlib.pyplot as plt
```
Comment (mhamilton723) [inline src/plot/src/main/python/plot.py]: Catch inport error

File: src/plot/src/main/python/plot.py [hunk 2]
```diff
@@ -13,12 +13,12 @@
 from pyspark.sql import DataFrame
 from pyspark.sql.types import *
```
Comment (mhamilton723) [inline src/plot/src/main/python/plot.py]: dont import star

File: src/plot/src/main/python/plot.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+from pyspark.ml.param.shared import *
+from pyspark.sql import DataFrame
+from pyspark.sql.types import *
+
+import numpy as np
+from sklearn.metrics import confusion_matrix, roc_curve
+import itertools
+
+def confusionMatrix(df, y_col, y_hat_col, labels):
+    import matplotlib.pyplot as plt
```
Comment (drdarshan) [inline src/plot/src/main/python/plot.py]:
```python
try:
  import matplotlib.pyplot as plt
except ImportError:
  raise Error("Please install matplotlib in order to use plotting") ##.. or some such thing.
```

File: src/udf/src/main/scala/UtilityUDFs.scala [hunk 4]
```diff
@@ -0,0 +1,28 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (mhamilton723) [inline src/udf/src/main/scala/UtilityUDFs.scala]: put in different package

File: src/readers/src/main/scala/ImageReader.scala [hunk 5]
```diff
@@ -97,4 +120,47 @@ object ImageReader {
...
+        val fs = path.getFileSystem(hconf.value)
+        val is = fs.open(path)
+        val imageRow = ImageReader.decode(path.toString,IOUtils.toByteArray(is)).getOrElse(Row(None))
+        val ret = Row.merge(Seq(row, Row(imageRow)):_*)
+        ret
+      }
+    }(encoder)
+  }
+
+  def write(df: DataFrame,
```
Comment (drdarshan) [inline src/readers/src/main/scala/ImageReader.scala]: Split into a Writer class?

File: src/udf/src/main/scala/UtilityUDFs.scala [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+// Licensed under the MIT License. See LICENSE in project root for information.
+
```
Comment (drdarshan) [inline src/udf/src/main/scala/UtilityUDFs.scala]: Move into a sub-package

File: src/plot/src/main/python/plot.py (hunk 2 above)
Comment (mhamilton723) [reply src/plot/src/main/python/plot.py]: done

File: src/udf/src/main/scala/UtilityUDFs.scala (hunk 6 above)
Comment (mhamilton723) [reply src/udf/src/main/scala/UtilityUDFs.scala]: made a udfs object. It is imported like a package.

File: src/plot/src/main/python/plot.py (hunk 3 above)
Comment (mhamilton723) [reply src/plot/src/main/python/plot.py]: done

File: src/plot/src/main/python/plot.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (drdarshan) [inline src/plot/src/main/python/plot.py]: Plotting should go into its own sub-module.. please file an issue if you are not able to refactor for this check-in.

File: src/plot/src/main/python/plot.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        plt.text(j, i, cm[i, j],
+                 horizontalalignment="center",
+                 fontsize=18,
+                 color="white" if cmn[i, j] > thresh else "black")
+
+    plt.colorbar()
+    plt.xlabel("Predicted Label", fontsize=18)
+    plt.ylabel("True Label", fontsize=18)
+
+def roc(df, y_col, y_hat_col, thresh=.5):
```
Comment (drdarshan) [inline src/plot/src/main/python/plot.py]: Like `sklearn.metrics`, I would split the creation of the ROC curve from the plotting itself.

File: src/readers/src/main/python/PowerBIWriter.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (drdarshan) [inline src/readers/src/main/python/PowerBIWriter.py]: Put it into a submodule so it doesn't looks like:
``py
import mmlspark;
mmlspark.PowerBIWriter(...)
``

File: src/readers/src/main/scala/PowerBIWriter.scala [hunk 10]
```diff
@@ -0,0 +1,101 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (drdarshan) [inline src/readers/src/main/scala/PowerBIWriter.scala]: Add a sub-package for visualization?

File: src/plot/src/main/python/plot.py (hunk 7 above)
Comment (mhamilton723) [reply src/plot/src/main/python/plot.py]: This is in its own submodule. If a python class is not explicitly imported in the __init__.py its addressable by the filename aka mmlspark.plot

File: src/plot/src/main/python/plot.py (hunk 8 above)
Comment (mhamilton723) [reply src/plot/src/main/python/plot.py]: I had wanted to provide a quick function that both computes and plots (though does not display allowing for composability and subplot embedding). If a user needed the extra control they could use the underlying sklearn.metrics function. Do you think that's reasonable?

File: src/readers/src/main/python/PowerBIWriter.py (hunk 9 above)
Comment (mhamilton723) [reply src/readers/src/main/python/PowerBIWriter.py]: Again this should look like mmlspark.PowerBIWriter.writeToPowerBI(....)
or df.writeToPowerBI(....)

**Pull request merged**